Add devolved_administration_availability to local_transaction schema

Rosa-Fox opened the pull request:
# What?

This schema has been added as part of a task to enable publishers to specify devolved administration (DA) service availability for local transactions. 

# Why?

In the Frontend application, by default, users will be shown a link to a local authority website when inputting a postcode for a local transaction.

In some cases, the service is handled by a devolved administration. In other cases the service is unavailable to devolved administrations.

Currently, the devolved administration availability cannot be set by publishers and the logic is defined in the Frontend application.

As part of the overall task, [functionality is being added to the Publisher application](https://github.com/alphagov/publisher/pull/1451) to allow this information to be set within the publishing tool.

Therefore, this schema validates the DA availability values set in Publisher.

Only the 'unavailable' and 'devolved_administration_service' related values will be presented from Publisher. This will be enough information to determine what to display on the Frontend as it will fall back to displaying the local authority results page.

# Useful links

Final naming decisions were agreed as a team during Dev chat on 26/05/2021. 

We changed the `type` value from `alternative_url` to `devolved_administration_service` as we felt it was confusing having `alternative_url` as a field name and as a `type` value. We also felt it better described the type of service.

[Modelling PR in Publisher](https://github.com/alphagov/publisher/pull/1456).

[Spike content schema branch](https://github.com/alphagov/govuk-content-schemas/pull/1060). 

[Spike write up](https://docs.google.com/document/d/1vOwGK_X4zUVuaMr4_KU4o8Jpfdn7dskqDq2cFjbQrBI/edit?usp=sharing).

[Spike PR for modelling in Publisher](https://github.com/alphagov/publisher/pull/1451).

[Trello](https://trello.com/c/Kgr3I4Tz/401-store-devolved-administration-local-transaction-data-in-model-and-govuk-content-schema).

Paired with @kevindew

leenagupte commented on dist/formats/local_transaction/frontend/schema.json:
```diff
@@ -349,15 +349,64 @@
         "need_to_know": {
           "$ref": "#/definitions/body_html_and_govspeak"
         },
+        "northern_ireland_availability": {
```
Are there separate "availability" values because it's possible to have more than one? Is it possible to have  a list of "devolved_administration_availability" entries in the content item, that have an enum for the country?

Rosa-Fox replied:
yeah there are more than one, it could be all 3.

leenagupte commented on dist/formats/local_transaction/frontend/schema.json:
```diff
@@ -349,15 +349,64 @@
+              "type": "string",
+              "format": "uri"
+            },
+            "type": {
+              "enum": [
+                "alternative_url"
+              ]
```
I think having "alternative_url" as the enum value and as a property is little confusing. I thought they were the same thing!

kevindew commented on formats/shared/definitions/devolved_administration_availability.jsonnet:
```diff
@@ -0,0 +1,26 @@
+{
+  devolved_administration_availability: {
+    description: "An object specifying the availability of service",
```
```suggestion
    description: "Used to indicate that a particular devolved administration has a different handling process for the service",
```

Rosa-Fox replied:
Thanks, I updated!

kevindew reviewed (commented):
This is looking good - I think you've got the syntax good now.

The few things I'd suggest are:

- split the generation of schema files into a separate commit (it's a common convention in this repo for example: https://github.com/alphagov/govuk-content-schemas/pull/1061/commits)
- Create an example schema that demonstrates the devolved administration fields (to pair with https://github.com/alphagov/govuk-content-schemas/blob/main/examples/local_transaction/frontend/local_transaction.json) this will allow us to test these schema changes against that example and potentially have something we can work with on the frontend in testing.

Rosa-Fox commented on formats/shared/definitions/devolved_administration_availability.jsonnet:
```diff
@@ -0,0 +1,26 @@
+{
```
thanks, I made the change.

Rosa-Fox commented:
Thanks @kevindew. I split up the commits and added an example file. The schema regeneration commit seems to be failing to build and I am not sure if it is related. I'll look at it first thing tomorrow.

kevindew commented on examples/local_transaction/frontend/local_transaction_with_devolved_administration_availability.json:
```diff
@@ -0,0 +1,331 @@
+    "northern_ireland_availability": {
+      "type": "devolved_administration_service",
+      "alternative_url": "https://www.nidirect.gov.uk/stray-dog"
+    },
```
Looks good 👍

kevindew reviewed (approved):
Nice one @Rosa-Fox
